package step03_multi_thread.thread05_synchronization.sync_method_and_block;

public class Calculator {
    private int memory;

    public int getMemory(){
        return memory;
    }

    //동기화 메소드
    public synchronized void setMemory1(int memory){
        this.memory = memory; //메모리 값 저장
        try{
            Thread.sleep(2000);
        }catch (InterruptedException e){
            //메모리 값 읽기
            System.out.println(Thread.currentThread().getName()+": "+this.memory);
        }
    }

    //동기화 블록
    public void setMemory2(int memory){
        synchronized (this){
            this.memory = memory;
            try{
                Thread.sleep(2000);
            }catch (InterruptedException e){
                //메모리 값 읽기
                System.out.println(Thread.currentThread().getName()+": "+this.memory);
            }
        }
    }
}
